### PR TITLE
Change the documentation search to use Algolia Docsearch

### PR DESCRIPTION
**Pull request by msh100**
This branch replaces the default sidebar search with Algolia Docsearch. Help by @davegri.

@lindseybateman does this visually look acceptable to you? Do you have any adjustments you'd like to make?

### CODE REVIEW

File: docs-articles/index.css [hunk 1]
```diff
@@ -36,7 +36,7 @@ body.sidebar-open .docs-articles {
   }
 
   .docs-articles {
-    z-index: 2;
+    z-index: 0;
```
Comment (akupila) [inline docs-articles/index.css]: Is `index: 0` needed or can you remove it entirely?

**Comment by akupila** [comment]
looks good to me! 👍

File: docs-articles/index.css (hunk 1 above)
Comment (msh100) [inline docs-articles/index.css]: I am not entirely sure. David's thought behind this one is that there will still be an index if it is required, just lower. 

I am not entirely sure why the index is there in the first place

Comment (akupila) [inline docs-articles/index.css]: @davegri ?

Probably not a huge deal to leave it in though

Comment (davegri) [inline docs-articles/index.css]: We couldn't find the reason but I gave it the benefit of the doubt and figured setting it to 0 was the minimum change needed to achieve our goal

**Comment by lindseybateman** [comment]
@msh100 pushed some styling changes to the Algolia output

**Comment by msh100** [comment]
@lindseybateman looks great, thanks!

**Comment by lindseybateman** [comment]
shall I merge it in @msh100 ?

**Comment by msh100** [comment]
Not yet, just want to double check this

**Comment by Tonnu** [comment]
@msh100 what's the status on this? The sooner we have it in the better :-)

**Pull request merged**